### PR TITLE
Fix deprecations in phase field module

### PR DESCRIPTION
**Pull request by cticenhour**
Started out fixing things related to deprecation failures in MALAMUTE, decided to do the rest of it while I am here. @dschwen would appreciate your input on some of my changes, since it went beyond simple find and replace in several commits. 

Refs #23260, #19155

### CODE REVIEW

File: modules/phase_field/include/kernels/CHBulk.h [hunk 1]
```diff
@@ -77,10 +77,8 @@ CHBulk<T>::validParams()
   InputParameters params = KernelGrad::validParams();
   params.addClassDescription("Cahn-Hilliard base Kernel");
   params.addParam<MaterialPropertyName>("mob_name", "M", "The mobility used with the kernel");
-  params.addDeprecatedCoupledVar("args",
-                                 "Vector of variable arguments of the mobility",
-                                 "args is deprecated, use 'coupled_variables' instead");
-  params.addCoupledVar("coupled_variables", "Vector of variable arguments of the mobility");
+  params.addCoupledVar("args", "Vector of variable arguments of the mobility");
+  params.deprecateCoupledVar("args", "coupled_variables", "02/27/2024");
```
Comment (dschwen) [inline modules/phase_field/include/kernels/CHBulk.h:81]: `coupled_variables` is the new not deprecated parameter.

Comment (dschwen) [reply modules/phase_field/include/kernels/CHBulk.h:81]: ok, so this is correct

**Review by dschwen — APPROVED** [review]
Thanks @cticenhour !

File: modules/phase_field/include/kernels/CHBulk.h (hunk 1 above)
Comment (cticenhour) [reply modules/phase_field/include/kernels/CHBulk.h:81]: Yup - the method arguments are `old_name`, `new_name`, and then `removal_date`.

**Comment by cticenhour** [comment]
Thanks for the review @dschwen!

**Pull request merged**